scripts/py_matter_yamltests/matter_yamltests/definitions.py

```diff
@@ -232,14 +242,29 @@ def __get_by_id(self, cluster_id: int, target_id: int, target_type: str):
 
         return targets.get(target_id)
 
-    def __enforce_casing(self, target_name: str, targets: list):
-        if targets.get(target_name) is not None:
-            return
+    def __get_targets_by_cluster_name(self, cluster_name: str, target_type: _ItemType):
+        if not cluster_name:
+            return None
+
+        # The idl parser remove spaces
+        cluster_name = cluster_name.replace(' ', '')
+
+        if target_type == _ItemType.Request:
+            targets = self.__commands_by_name.get(cluster_name)
+        elif target_type == _ItemType.Response:
+            targets = self.__responses_by_name.get(cluster_name)
+        elif target_type == _ItemType.Event:
+            targets = self.__events_by_name.get(cluster_name)
+        elif target_type == _ItemType.Attribute:
+            targets = self.__attributes_by_name.get(cluster_name)
+        elif target_type == _ItemType.Bitmap:
+            targets = self.__bitmaps_by_name.get(cluster_name)
+        elif target_type == _ItemType.Enum:
+            targets = self.__enums_by_name.get(cluster_name)
+        elif target_type == _ItemType.Struct:
+            targets = self.__structs_by_name.get(cluster_name)
```

Comment:
should there be a final elseif saying `unsupported item type` ?

Also how about early return like:

```py
if target_type == _ItemType.Request:
    return self.__commands_by_name.get(cluster_name)
if target_type == _ItemType.Response:
    return self.__responses_by_name.get(cluster_name)
# ...
raise Exception("Unsupported type: %r" % target_type)
```

or mapping wise (not tested, but looks like it may work assuming ItemType is hashable ... and would raise key error if key does not exist):

```py
target_mapping = {
   _ItemType.Request: self.__commands_by_name,
   _ItemType.Response: self.__responses_by_name,
   _ItemType.Event: self.__events_by_name,
   #  ...
}

return target_mapping[target_type].get(cluster_name)
```

Reply:
I have updated the code to be "mapping wise" as you suggest.